spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/ApiClient.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.bootstrap.client;
+
+import org.candlepin.resource.ActivationKeyApi;
+import org.candlepin.resource.AdminApi;
+import org.candlepin.resource.CdnApi;
+import org.candlepin.resource.ConsumerTypeApi;
+import org.candlepin.resource.OwnerApi;
+import org.candlepin.resource.PoolsApi;
+import org.candlepin.resource.ProductsApi;
+import org.candlepin.resource.RolesApi;
+import org.candlepin.resource.RootApi;
+import org.candlepin.resource.StatusApi;
+import org.candlepin.resource.UsersApi;
+import org.candlepin.spec.bootstrap.client.api.ConsumerAPI;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import java.util.Objects;
+
+/**
+ * An entrypoint of Candlepin client. Serves as a crossroads between specific APIs.
+ */
+public class ApiClient {
```

Comment:
I'm a little confused on why we need this wrapper class, and the naming is confusing as well considering it is org.candlepin.spec.bootstrap.client.ApiClient that wraps the org.candlepin.ApiClient.

Reply:
The purpose is to handle creation of different APIs and sharing an instance of okhttp client between them. Naming could be improved. I'll try to figure something out but I am open to suggestions.